### PR TITLE
Http config doc v1

### PR DESCRIPTION
**Pull request by catenacyber**
Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
https://redmine.openinfosecfoundation.org/issues/3207

Describe changes:
- Updates suricata.yaml template with more comments about every keyword about HTTP parsing
- Updates Userguide about configuration of suricata.yaml to have the good list of keywords about HTTP parsing

### CODE REVIEW

File: suricata.yaml.in [hunk 1]
```diff
@@ -896,17 +897,40 @@ app-layer:
...
+      #                           Default is "preserve_percent"
+      #   path-utf8-convert-bestfit:
+      #                           Decodes unicode, and replaces invalid characters with bestfit
+      #                           Default is true with IDS personality
+      #   path-bestfit-replacement-char:
+      #                           Selects character to replace invalid unicode
+      #                           Default is '?'
+      #   query-plusspace-decode:
+      #                           Decodes plus as space
+      #                           Default is true
```
Comment (victorjulien) [inline suricata.yaml.in]: I think that these, and probably a bunch of the existing options, should not be in the default config. Instead we should have a comment like "for advanced options see XYZ", where XYZ a link to the user guide.

File: suricata.yaml.in [hunk 2]
```diff
@@ -975,10 +1004,13 @@ app-layer:
...
 
+           # Can disable LZMA decompression
            #lzma-enabled: yes
-           # LZMA decompression memory limit.
+           # Memory limit usage for LZMA decompression dictionary
+           # Data is decompressed until dictionary reaches this size
            #lzma-memlimit: 1 Mb
-           # Compression bomb output limit.
+           # Maximum decompressed size with a compression ratio
+           # above 2048 (only reachable by LZMA)
```
Comment (victorjulien) [inline suricata.yaml.in]: "only reachable by LZMA" took me a few reads to understand. Perhaps update to "only LZMA can reach this ratio" ?

File: doc/userguide/configuration/suricata-yaml.rst [hunk 3]
```diff
@@ -1316,12 +1316,12 @@ The library Libhtp is being used by Suricata to parse HTTP-sessions.
 While processing HTTP-traffic, Suricata has to deal with different
 kind of servers which each process anomalies in HTTP-traffic
 differently. The most common web-server is Apache. This is a open
```
Comment (norg) [inline doc/userguide/configuration/suricata-yaml.rst]: is a**n** open

File: doc/userguide/configuration/suricata-yaml.rst [hunk 4]
```diff
@@ -1316,12 +1316,12 @@ The library Libhtp is being used by Suricata to parse HTTP-sessions.
...
-source web -server program.
+source web-server program.
 
-Beside Apache, IIS (Internet Information Services/Server)a web-server
+Beside Apache, IIS (Internet Information Services/Server) a web-server
```
Comment (norg) [inline doc/userguide/configuration/suricata-yaml.rst]: Beside**s**

File: suricata.yaml.in (hunk 2 above)
Comment (catenacyber) [reply suricata.yaml.in]: ok

File: suricata.yaml.in (hunk 1 above)
Comment (catenacyber) [reply suricata.yaml.in]: ok

File: doc/userguide/configuration/suricata-yaml.rst (hunk 3 above)
Comment (catenacyber) [reply doc/userguide/configuration/suricata-yaml.rst]: thanks

File: doc/userguide/configuration/suricata-yaml.rst (hunk 4 above)
Comment (catenacyber) [reply doc/userguide/configuration/suricata-yaml.rst]: thanks

**Pull request closed without merging**